test-suite/Math/NumberTheory/ArithmeticFunctions/InverseTests.hs

```diff
@@ -132,15 +151,74 @@ totientSpecialCases3 = zipWith mkAssert (tail factorial) totientMaxFactorial
     totientMax :: Word -> Word
     totientMax = unMaxWord . inverseTotient MaxWord
 
+jordans5 :: [Word]
+jordans5 =
+  [ 1
+  , 31
+  , 242
+  , 992
+  , 3124
+  , 7502
+  , 16806
+  , 31744
+  , 58806
+  , 96844
+  , 161050
+  , 240064
+  , 371292
+  , 520986
+  , 756008
+  , 1015808
+  , 1419856
+  , 1822986
+  , 2476098
+  , 3099008
+  , 4067052
+  , 4992550
+  , 6436342
+  , 7682048
+  , 9762500
+  , 11510052
+  , 14289858
+  , 16671552
+  , 20511148
+  ]
+
+jordanSpecialCase1 :: [Assertion]
+jordanSpecialCase1 = zipWith mkAssert ixs jordans5
+  where
+    mkAssert a b = assertEqual "should be equal" (S.singleton a) (asSetOfPreimages (inverseJordan 5) b)
+    ixs = [1 .. 29]
+
 -------------------------------------------------------------------------------
 -- Sigma
 
 sigmaProperty1 :: forall a. (Euclidean a, UniqueFactorisation a, Integral a, Enum (Prime a), Bits a) => Positive a -> Bool
 sigmaProperty1 (Positive x) = x `S.member` asSetOfPreimages inverseSigma (sigma 1 x)
 
+sigmaKProperty1
+  :: forall a
+   . (Euclidean a, UniqueFactorisation a, Integral a, Enum (Prime a), Bits a)
+  => Power Word
+  -> Positive a
+  -> Bool
+sigmaKProperty1 (Power k') (Positive x) =
+  -- 'k' shouldn't be large to avoid slow tests.
+  let k = 2 + k' `Prelude.mod` 20
```

Comment:
You use `Prelude.mod` here and below, but `Prelude.div` for `jordanProperty{1,2}`.

Reply:
Nice find: meant to be `mod` in both places.